File: sort/选择排序/section.java
/* 
    input:
        4
        1
        5
        2
        3
    output:
        1
        2
        3
        4
        5
*/
public class section {
    public static void main(String []args){
        int[] array = {4,1,5,2,3};
        System.out.println("input:");
        for(int a:array)
            System.out.println(a);
        System.out.println("output:");
        for(int b:sectionSort(array))
            System.out.println(b);
    }
    // 排序
    public static int[] sectionSort(int []arr){
        for(int i =0;i<arr.length-1;i++) { 
            int min = i;
            for(int j=i+1;j<arr.length;j++) {
                if(arr[j]<arr[min]) {
                    min = j;
                }
            }    
            int temp = arr[min];
            arr[min]=arr[i];
            arr[i]=temp;
        }
        return arr;
    }
}

File: sort/选择排序/section.cpp
# include<iostream>
using namespace std;

/*
	input:
	4 1 5 2 4
	output:
	1 2 4 4 5
*/
void Section(int array[],int len){
	int i,j,temp,min;
    for (i = 0;i<len-1;i++)
    {
        min = i;
        for (j = i+1; j < len; j++)
        {
            if (array[j] < array[min])
                min = j;
        }
        temp = array[min];
        array[min] = array[i];
        array[i] = temp;
    }
}

int main(){
	int array[]={4,1,5,2,3}; 
	int len = sizeof(array)/sizeof(array[0]);
	cout << "input:" << endl;
	for (int i = 0; i < len; i++)
        cout << array[i] << ' ';
	cout << endl;
	// section sort 
	Section(array,len);
	cout << "output:" << endl;
	for (int i = 0; i < len; i++)
        cout << array[i] << ' ';
	cout << endl;
}
